projects/project3/src/main/scala/blast/main/Main.scala

```diff
@@ -0,0 +1,94 @@
+package blast.main
+
+import org.apache.spark.sql.SparkSession
+import org.apache.jena.riot.Lang
+import org.apache.spark.rdd.RDD
+import org.apache.hadoop.hdfs.DFSClient
+import DataStructures.{Attribute, DatasetReader, EntityProfile, IdDuplicates}
+import blast.AttributeSchema.AttributeProfile
+import blast.AttributeSchema.AttributeMatchInduction
+import blast.Blocking.Blocker
+import blast.Blocking.MetaBlocker
+import blast.data_processing.read_GroundTruth
+
+import scala.collection.JavaConverters._
+
+
+
+
+object Main {
+
+  def convertFile(spark : SparkSession, inputP:String, outputP: String): Unit = {
+    val ds:RDD[EntityProfile] = spark.sparkContext.parallelize(DatasetReader.readDataset(inputP))
+    ds.saveAsObjectFile(outputP)
+  }
+  def main(args: Array[String]) {
+    //initializing spark
+    val spark = SparkSession.builder
+      .appName(s"Blast")
+      .master("local[*]")
+      .getOrCreate()
+
+    val ds1suffix = "dblp"
+    val ds2suffix = "acm"
+
+    val ds1path = "/media/sf_uniassignments/BLAST/dataset1_"+ ds1suffix
+    val ds1pathScala = ds1path.concat("_scala")
+    val ds2path = "/media/sf_uniassignments/BLAST/dataset2_" + ds2suffix
+    val ds2pathScala = ds2path.concat("_scala")
+
+    // for the first run, uncomment this part to format the files into spark format
+    //convertFile(spark, ds1path, ds1pathScala)
+    //convertFile(spark, ds2path, ds2pathScala)
+
+    //return
+    //**read dataset with spark, should use the old method first to read the data for the first time
+    val dataS1 :RDD[EntityProfile] = spark.sparkContext.objectFile(ds1pathScala)
```

Comment:
I see that you did use SparkReader, but again the data are first to read using JavaStream. We could discuss this during your presentation, as there is no much time left before the final presentation.